### PR TITLE
[Collab] Collaboration service skeleton

### PR DESCRIPTION
**Pull request by jeffrey-jian**
This pull request introduces the new `collaboration-service` to the project. 

The main changes include adding new nestjs controller, service, and related configurations for the collaboration-service and setting up Redis and Supabase integrations.
 
View architecture diagram here:
https://drive.google.com/file/d/1d7s2ey8xPaVSpilghh2N2jdUDiBrhSHK/view?usp=sharing

### CODE REVIEW

File: project/apps/collaboration-service/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+async function bootstrap() {
+  const appContext =
+    await NestFactory.createApplicationContext(CollaborationModule);
+  const envService = appContext.get(EnvService);
+  const NODE_ENV = envService.get('NODE_ENV');
+  const COLLABORATION_SERVICE_HOST = envService.get(
+    'COLLABORATION_SERVICE_HOST',
+  );
+  appContext.close();
```
Comment (javinchua) [inline project/apps/collaboration-service/src/main.ts:16]: prob dw to do this similar to matching-service

Comment (jeffrey-jian) [reply project/apps/collaboration-service/src/main.ts]: oops you're right, i missed out a commit while doing `git cherry-pick`.

amended.

File: project/apps/collaboration-service/src/db/collaboration.supabase.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
+import { createClient, SupabaseClient } from '@supabase/supabase-js';
+import { EnvService } from 'src/env/env.service';
+// import { firstValueFrom } from 'rxjs';
+// import { ClientProxy } from '@nestjs/microservices';
+
+@Injectable()
+export class MatchSupabase {
```
Comment (javinchua) [inline project/apps/collaboration-service/src/db/collaboration.supabase.ts:8]:
```suggestion
export class CollabSupabase {
```

**Review by javinchua — APPROVED** [review]
its quite messy as of now but ill just approve first

**Pull request merged**